app/models/org_space_policy.rb

```diff
@@ -46,18 +48,24 @@ def create
   def delete
     raise InstanceNotFound unless instance_resource.exists?
   
-    load_policy(template_delete_space, 
-      method: Conjur::API::POLICY_METHOD_PATCH)
+    unless preserve_policy?
+      load_policy(template_delete_space, 
+        method: Conjur::API::POLICY_METHOD_PATCH)
 
-    load_policy(template_delete_org, 
-      method: Conjur::API::POLICY_METHOD_PATCH) if org_empty?(instance_org_id)
+      load_policy(template_delete_org, 
+        method: Conjur::API::POLICY_METHOD_PATCH) if org_empty?(instance_org_id)
+    end
 
     load_policy(template_delete_instance, 
       method: Conjur::API::POLICY_METHOD_PATCH)
   end
 
   private
 
+  def preserve_policy?
+    ENV['CONJUR_PRESERVE_POLICY'].to_boolean
```

Comment:
I think I agree with @jtuttle. It starts to add complexity when you think about asking users to configure the tile with a boolean parameter as well - it may be easier to use a string.

If we decide to require a string "true" to turn this on, the issue in conjurinc/cloudfoundry-conjur-tile#14 will also need to be updated to reflect this change in thinking.

Reply:
> I've always considered monkey-patching to be a code smell that should be avoided whenever possible.
Sure, I can find an alternative way to do it.

> Can we not just do `ENV['CONJUR_PRESERVE_POLICY'] == "true"` and make it explicit through documentation that this needs to be set to `true` rather than supporting a variety of truthy values?
I think this actually important for getting to that love-able, "just works™️" experience. Sure, we can document the best-practice, but not being flexible on something that isn't consistent everywhere is just asking for end-user pain.

I know it seems pedantic to make the argument on this particular setting, but this is the kind of end-user rigidity that is pervasive in Conjur. We can't be flexible everywhere, but something like string to boolean is an easy opportunity for a UX win.

> It starts to add complexity when you think about asking users to configure the tile with a boolean parameter as well - it may be easier to use a string.
I don't have an opinion on this other than, if the user input is a text box, I think it's even more important that we're flexible in how we convert it to the boolean value.

Reply:
it turns out the form does allow boolean input (appears as a checkbox): https://docs.pivotal.io/tiledev/2-0/product-template-reference.html#configurable-boolean

we can test this out by generating a test tile and deploying it to our PCF via the ops manager (to see what the form looks like, and what the service broker receives in its config)